apps/expo-go/ios/Exponent/Kernel/AppLoader/EXAppLoaderExpoUpdates.m

```diff
@@ -52,6 +52,9 @@ @interface EXAppLoaderExpoUpdates ()
 @property (nonatomic, nullable) EXUpdatesSelectionPolicy *selectionPolicy;
 @property (nonatomic, nullable) id<EXUpdatesAppLauncher> appLauncher;
 
+@property (nonatomic, nullable) NSDate *startupStartTime;
+@property (nonatomic, nullable) NSDate *startupEndTime;
```

Comment:
maybe we can use `dispatch_time_t`, given we don't need year, month and so on.

```objc
dispatch_time_t now = dispatch_time(DISPATCH_TIME_NOW, 0);
```

Reply:
I originally had that but opted to use a wrapped type so that it could be nullable so that we could know more easily when it should be a null return value for `launchDuration`. Otherwise we'd have to keep 4 pieces of information:
- `dispatch_time_t startupStartTime`
- `BOOL didSetStartupStartTime`
- `dispatch_time_t startupEndTime`
- `BOOL didSetStartupEndTime`